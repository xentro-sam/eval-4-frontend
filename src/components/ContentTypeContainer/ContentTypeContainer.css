#content-type-container {
    box-sizing: border-box;
    padding: 1vw 3vw;
}

#add-content-type-button {
    border: 2px dotted rgb(78, 34, 198);
    color: rgb(78, 34, 198);
    font-size: 1vw;
    font-weight: 600;
    border-radius: 5px;
    padding: 1vw 2vw;
    background-color: rgb(233, 234, 241);
    text-align: center;
}