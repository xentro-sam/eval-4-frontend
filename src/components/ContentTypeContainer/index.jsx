import * as React from 'react';
import './ContentTypeContainer.css';
import makeRequest from '../../utils/makeRequest';
import {GET_CONTENT_TYPES} from '../../constants/apiEndPoints';
import ContentType from '../ContentType';
import PropTypes from 'prop-types';

export default function ContentTypeContainer(props) {
  const [contentTypes, setContentTypes] = React.useState([]);
  React.useEffect(() => {
    makeRequest(GET_CONTENT_TYPES, {})
        .then((response) => {
          setContentTypes(response);
          props.setCollectionTypes(response);
        });
  }, []);
  return (
    <div id="content-type-container">
      <div id="content-type-container-search">

      </div>
      <div id="content-type-container-content">
        <div id="add-content-type-button">
            + New Type
        </div>
        <div id="content-type-container-content-list">
          {contentTypes.map((contentType) => {
            return (
              <ContentType key={contentType.id} {...contentType} setFields={props.setFields} setContainerTitle={props.setContainerTitle} />
            );
          })}
        </div>
      </div>
    </div>
  );
}

ContentTypeContainer.propTypes = {
  setFields: PropTypes.func.isRequired,
  setContainerTitle: PropTypes.func.isRequired,
  setCollectionTypes: PropTypes.func.isRequired,
};
