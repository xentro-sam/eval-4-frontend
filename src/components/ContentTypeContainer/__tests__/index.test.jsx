import React from 'react';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ContentTypeContainer from '../index';
import {waitFor} from '@testing-library/dom';
import makeRequest from '../../../utils/makeRequest';
import {act} from 'react-dom/test-utils';

jest.mock('../../../utils/makeRequest');

const mockData = {
  setFields: jest.fn(),
  setContainerTitle: jest.fn(),
  setCollectionTypes: jest.fn(),
  setContentTypeId: jest.fn(),
};

const mockResponse = [{
  id: 1,
  contentTypeName: 'mock title 1',
}];

const mockPromise = Promise.resolve(mockResponse);

describe('ContentType', () => {
  it('should render correctly', async () => {
    makeRequest.mockImplementation(() => mockPromise);
    expect(makeRequest).not.toHaveBeenCalled();
    const {asFragment} = await act( async () => render(<ContentTypeContainer {...mockData} />));
    await waitFor(() => {
      expect(screen.getByText('+ New Type')).toBeTruthy();
    });
    expect(asFragment()).toMatchSnapshot();
    expect(makeRequest).toBeCalled();
  });
});
