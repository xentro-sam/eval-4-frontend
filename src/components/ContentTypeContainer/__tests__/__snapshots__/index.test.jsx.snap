// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentType should render correctly 1`] = `
<DocumentFragment>
  <div
    id="content-type-container"
  >
    <div
      id="content-type-container-search"
    />
    <div
      id="content-type-container-content"
    >
      <div
        id="add-content-type-button"
      >
        + New Type
      </div>
      <div
        id="content-type-container-content-list"
      >
        <div
          id="content-type"
        >
          <div
            id="content-type-name"
          >
            mock title 1
          </div>
          <div
            id="content-type-count"
          >
            1
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
